beefy-gadget/src/worker.rs

```diff
@@ -313,6 +317,14 @@ where
 
 				debug!(target: "beefy", "🥩 Round #{} concluded, committed: {:?}.", round.1, signed_commitment);
 
+				if self
+					.backend
+					.append_justification(BlockId::Number(round.1), (BEEFY_ENGINE_ID, signed_commitment.encode()))
```

Comment:
I think it makes sense to introduce one-byte versioning right from start.

Reply:
So this would be something like:
```
let version: u8;
... (BEEFY_ENGINE_ID, (version, signed_commitment).encode())
```
?

Reply:
Yeah, or just:
```
#[derive(Encode, Decode)]
pub enum VersionedSignedCommitment {
  V1(SignedCommitment),
}
```

which should be easier to work with long term - the resulting encoding though should be identical in any of the solutions afair though.